test/library/metrics/query.go

```diff
@@ -0,0 +1,88 @@
+package metrics
+
+import (
+	"crypto/tls"
+	"fmt"
+	"net"
+	"net/http"
+	"strings"
+	"time"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/transport"
+
+	prometheusapi "github.com/prometheus/client_golang/api"
+	prometheusv1 "github.com/prometheus/client_golang/api/prometheus/v1"
+
+	routeclient "github.com/openshift/client-go/route/clientset/versioned"
+)
+
+// NewPrometheusClient returns Prometheus API or error
+// Need to distinguish between a query or an alert, since for querying one should
+// access the thanos-querier route but for alerts access the prometheus-k8s route.
+// Hence, need to pass 'queryType' as either 'query' or 'alert'
+func NewPrometheusClient(kclient *kubernetes.Clientset, rc *routeclient.Clientset, queryType string) (prometheusv1.API, error) {
+	_, err := kclient.CoreV1().Services("openshift-monitoring").Get("prometheus-k8s", metav1.GetOptions{})
+	if err != nil {
+		return nil, err
+	}
+
+	var host string
+	switch queryType {
+	case "query":
+		route, err := rc.RouteV1().Routes("openshift-monitoring").Get("thanos-querier", metav1.GetOptions{})
+		if err != nil {
+			return nil, err
+		}
+		host = route.Status.Ingress[0].Host
+	case "alert":
+		route, err := rc.RouteV1().Routes("openshift-monitoring").Get("prometheus-k8s", metav1.GetOptions{})
```

Comment:
@s-urbaniak If this route is deprecated, how should tests be querying for alerts?

Reply:
^^ lol I found that out also, see above

Reply:
as mentioned above you can use the `ALERTS` query to get the desired result. Indeed, the `/alerts` endpoint is not accessible via Thanos Querier, as it is "just" responsible for querying metrics.